scripts/tile_benchmark/benchmark.py

```diff
@@ -0,0 +1,365 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#############################################################################
+
+
+import math
+import sys
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+
+np.set_printoptions(threshold=np.nan)
+
+URL = 0
+NUM_LEVELS = 1
+NUM_X_TILES = 2
+NUM_Y_TILES = 3
+ATLAS_ARR = 4
+
+
+def main():
+    """Read in request data from source.txt, request tiles from each server and
+    visualize response times
+    """
+    file_name = 'source.txt'
+    run_params = {}
+    # read in servers to request
+    with open(file_name, 'r') as params:
+        for line in params:
+            if '#' not in line:
+                args = line.split()
+                run_params[args[0]] = args
+    # get elapsed time for data requests
+    for host in run_params.keys():
+        if host == 'atlas':
+            print "Getting images from atlas"
+            elapsed_time = get_data(
+                "atlas",
+                'atlas hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        elif host == 'adrc':
+            print "Getting images from adrc"
+            elapsed_time = get_data(
+                "adrc",
+                'adrc hardcoded img',
+                int(run_params[host][1]),
+                int(run_params[host][2]))
+        else:
+            print"Getting images from girder"
+            elapsed_time = get_data(
+                run_params[host][0],
+                run_params[host][1],
+                int(run_params[host][2]),
+                int(run_params[host][3]))
+        run_params[host].append(elapsed_time)
+
+    plot_data_and_save(run_params)
+
+
+def get_data(host, image_id, requested_levels, requested_num_tiles):
+    """get request information from each server an request tiles"""
+    num_levels = -1
+    data = {}
+    # get data about server, host url and params
+    if host == "atlas":
+
+        data = atlas_init()
+    elif host == "adrc":
+
+        data = adrc_init()
+    else:
+        data = girder_init(host, image_id)
+    num_levels = data[NUM_LEVELS]
+    # if successful begin to grab tiles from each level
+    if num_levels != -1:
+        if requested_levels > num_levels:
+            requested_levels = num_levels
+        # get elapsed time info numpy array include
+        elapsed_time_arr, count = \
+            iterate_over_tiles(
+                host,
+                data,
+                requested_levels,
+                requested_num_tiles)
+
+        # create a new array of correct size and plot
+
+        return elapsed_time_arr
+
+
+def iterate_over_tiles(host, data, requested_levels, requested_num_tiles):
+    """iterate from level 0 to higher levels requesting tiles"""
+    total_tiles = 0
```

Comment:
It'd be cleaner to rewrite this as a comprehension with `sum`.

Or there's probably an analytic solution to the power series. :wink: